packages/ui/components/avatar/avatar.test.tsx

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/* eslint-disable playwright/missing-playwright-await */
+import { render, screen, waitFor, act, fireEvent } from "@testing-library/react";
+import { useState } from "react";
+import { beforeAll, vi } from "vitest";
+
+import { Avatar, sizesPropsBySize } from "./Avatar";
+import { AvatarGroup } from "./AvatarGroup";
+
+type sizeKeys = "xxs" | "xs" | "xsm" | "sm" | "md" | "mdLg" | "lg" | "xl";
+
+vi.mock("@radix-ui/react-avatar", async () => {
+  const ImageMock = (props: { [key: string]: string; alt: string }) => {
+    return <img {...props} alt={props.alt} />;
+  };
+  const actual = (await vi.importActual("@radix-ui/react-avatar")) as { default: object };
+  return {
+    ...actual.default,
+    Image: ImageMock,
+  };
+});
+
+const observeMock = vi.fn();
+
+window.ResizeObserver = vi.fn().mockImplementation(() => ({
+  disconnect: vi.fn(),
+  observe: observeMock,
+  unobserve: vi.fn(),
+}));
+
+vi.mock("../tooltip", async () => {
+  const actual = (await vi.importActual("../tooltip")) as any;
+  const TooltipMock = (props: object) => {
+    const [open, setOpen] = useState(false);
+
+    return (
+      <actual.Tooltip
+        {...props}
+        open={open}
+        onOpenChange={(isOpen: boolean) => {
+          setOpen(isOpen);
+        }}
+      />
+    );
+  };
+  return {
+    Tooltip: TooltipMock,
+  };
+});
+
```

Comment:
All these mocks are for Tooltip testing, radix uses ResizeObserver which js-dom doesn't have,
so I'm simulating its behavior, as it's just a mock, radix doesn't actually render the tooltip content in the resulting HTML,
but in this way, I managed to prove that the event is detected in the component and that once the state is updated, 
the `data-state` attribute, which comes from the radix, is modified internally (`data-state` starts as `closed `and ends as `instant-open`, see test block)